GPU available: False, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
/home/user126/anaconda3/envs/dt_idl/lib/python3.11/site-packages/torch/nn/init.py:412: UserWarning: Initializing zero-element tensors is a no-op
  warnings.warn("Initializing zero-element tensors is a no-op")
/home/user126/anaconda3/envs/dt_idl/lib/python3.11/site-packages/pytorch_lightning/trainer/configuration_validator.py:108: PossibleUserWarning: You defined a `validation_step` but have no `val_dataloader`. Skipping val loop.
  rank_zero_warn(
  | Name                       | Type                       | Params
--------------------------------------------------------------------------
0 | A_input_transformation     | Linear                     | 80
1 | X_input_transformation     | Linear                     | 16
2 | Y_input_transformation     | Linear                     | 32
3 | V_input_transformation     | Linear                     | 32
4 | self_positional_encoding_k | RelativePositionalEncoding | 248
5 | self_positional_encoding_v | RelativePositionalEncoding | 248
6 | transformer_blocks         | ModuleList                 | 8.2 K
7 | output_dropout             | Dropout                    | 0
--------------------------------------------------------------------------
8.4 K     Trainable params
0         Non-trainable params
8.4 K     Total params
0.033     Total estimated model params size (MB)
/home/user126/anaconda3/envs/dt_idl/lib/python3.11/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:224: PossibleUserWarning: The dataloader, train_dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 32 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(
/home/user126/anaconda3/envs/dt_idl/lib/python3.11/site-packages/torch/nn/_reduction.py:42: UserWarning: size_average and reduce args will be deprecated, please use reduction='none' instead.
  warnings.warn(warning.format(ret))
































































































































































































































































































































































































































































































































































































































































































































Epoch 149:  90%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                  | 36/40 [00:08<00:00,  4.27it/s, loss=0.123, v_num=9wfo]
`Trainer.fit` stopped: `max_epochs=150` reached.
/home/user126/anaconda3/envs/dt_idl/lib/python3.11/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:224: PossibleUserWarning: The dataloader, test_dataloader 0, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 32 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(
/home/user126/TFformer/model/CT_ourmodel.py:347: UserWarning: Using a target size (torch.Size([256, 59, 1])) that is different to the input size (torch.Size([256, 59, 16])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
Epoch 149: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [00:09<00:00,  4.36it/s, loss=0.122, v_num=9wfo]
Testing DataLoader 0: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 12.51it/s]
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Runningstage.testing metric      DataLoader 0
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
      test_mse_loss          6.755414063599461
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Predicting DataLoader 0: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 12.69it/s]
outcome_pred 4
Predicting DataLoader 0: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 12.52it/s]
Predicting DataLoader 0:   3%|██████▌                                                                                                                                                                                           | 3/89 [00:00<00:05, 14.86it/s]
/home/user126/TFformer/model/CT_ourmodel.py:347: UserWarning: Using a target size (torch.Size([232, 59, 1])) that is different to the input size (torch.Size([232, 59, 16])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  mse_loss = nn.functional.mse_loss(outcome_pred, batch['outputs'], reduce=False)
INFO:model.CT_ourmodel:RMSE calculation for val.
INFO:model.CT_ourmodel:Predictions for val.
/home/user126/anaconda3/envs/dt_idl/lib/python3.11/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:224: PossibleUserWarning: The dataloader, predict_dataloader 0, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 32 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(
INFO:__main__:Val normalised RMSE (all): 49.80689594459902; Val normalised RMSE (orig): 49.68421620388407
INFO:model.CT_ourmodel:RMSE calculation for test.


Predicting DataLoader 0: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 89/89 [00:06<00:00, 14.23it/s]
outcome_pred 89



Predicting DataLoader 0: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 89/89 [00:05<00:00, 14.91it/s]
INFO:__main__:Test normalised RMSE (all): 64.20309881961987; Test normalised RMSE (orig): 55.97235383851731; Test normalised RMSE (only counterfactual): 56.833765794703886
{'encoder_val_rmse_all': 49.80689594459902, 'encoder_val_rmse_orig': 49.68421620388407, 'encoder_test_rmse_all': 64.20309881961987, 'encoder_test_rmse_orig': 55.97235383851731, 'encoder_test_rmse_last': 56.833765794703886}